src/drivers/intel/ssp/ssp.c

```diff
@@ -613,35 +613,30 @@ static int ssp_set_config(struct dai *dai, struct ipc_config_dai *common_config,
 	return ret;
 }
 
-/*
- * Portion of the SSP configuration should be applied just before the
- * SSP dai is activated, for either power saving or params runtime
- * configurable flexibility.
- */
-static int ssp_pre_start(struct dai *dai)
+static int ssp_set_mclk(struct dai *dai)
 {
 	struct ssp_pdata *ssp = dai_get_drvdata(dai);
 	struct sof_ipc_dai_config *config = &ssp->config;
-	uint32_t sscr0;
-	uint32_t mdiv;
-	bool need_ecs = false;
-
-	int ret = 0;
-
-	dai_info(dai, "ssp_pre_start()");
-
-	/* SSP active means bclk already configured. */
-	if (ssp->state[SOF_IPC_STREAM_PLAYBACK] == COMP_STATE_ACTIVE ||
-	    ssp->state[SOF_IPC_STREAM_CAPTURE] == COMP_STATE_ACTIVE)
-		return 0;
+	int ret;
 
 	/* MCLK config */
 	ret = mn_set_mclk(config->ssp.mclk_id, config->ssp.mclk_rate);
-	if (ret < 0) {
+	if (ret < 0)
 		dai_err(dai, "invalid mclk_rate = %d for mclk_id = %d",
 			config->ssp.mclk_rate, config->ssp.mclk_id);
-		goto out;
-	}
+
+
+	return ret;
+}
+
+static int ssp_set_bclk(struct dai *dai)
+{
+	struct ssp_pdata *ssp = dai_get_drvdata(dai);
+	struct sof_ipc_dai_config *config = &ssp->config;
+	uint32_t sscr0;
+	uint32_t mdiv;
+	bool need_ecs = false;
+	int ret = 0;
```

Comment:
no need to init?

Reply:
> no need to init?

We need to init because `ret = mn_set_bclk()` is only called if  `CONFIG_INTEL_MN`